test/val/val_decoration_test.cpp

```diff
@@ -146,5 +146,37 @@ TEST_F(ValidateDecorations, ValidateGroupMemberDecorateRegistration) {
   EXPECT_THAT(vstate_->id_decorations(4), Eq(expected_decorations));
 }
 
+TEST_F(ValidateDecorations, LinkageImportUsedForInitializedVariableBad) {
+  string spirv = R"(
+               OpCapability Shader
+               OpCapability Linkage
+               OpMemoryModel Logical GLSL450
+               OpDecorate %target LinkageAttributes "link_ptr" Import
+      %float = OpTypeFloat 32
+ %_ptr_float = OpTypePointer Uniform %float
+      %var_1 = OpVariable %_ptr_float Input
+     %target = OpVariable %_ptr_float Uniform %var_1
```

Comment:
The intializer for a float variable should be a float value.  But %var_1 is itself a pointer to float.
So this test would fail for the wrong reason if we had type checking for initializers.

Suggest that instead of %var_1 you write:

%zero = OpConstantNull %float

Then make %zero your initializer value.